clarity-ext-scripts/tests/test_rtpcr_analysis_service.py

```diff
@@ -204,3 +204,66 @@ def test_can_analyze_samples_scenario6(self):
             {"id": "cov-3", "diagnosis_result": FAILED_ENTIRE_PLATE_BY_FAILED_EXTERNAL_CONTROL}]
 
         assert expected == list(result)
+
+    def test_can_analyze_samples_scenario7(self):
+        """
+        For no signal on both channels of a negative control, it should be negative
+        """
+
+        pos_controls = [{"id": "pos-1", "FAM-CT": 25, "HEX-CT": 20},
+                        {"id": "pos-2", "FAM-CT": 20, "HEX-CT": 18}]
+
+        neg_controls = [{"id": "neg-1", "FAM-CT": 0, "HEX-CT": 0},
+                        {"id": "neg-2", "FAM-CT": 0, "HEX-CT": 0}]
+
+        samples = [{"id": "cov-1", "FAM-CT": 20, "HEX-CT": 20},
+                   {"id": "cov-2", "FAM-CT": 14, "HEX-CT": 18},
+                   {"id": "cov-3", "FAM-CT": 40, "HEX-CT": 33}]
+
+        result = self.service.analyze_samples(positive_controls=pos_controls,
+                                              negative_controls=neg_controls,
+                                              samples=samples)
+
+        expected = [
+            {"id": "pos-1", "diagnosis_result": COVID_RESPONSE_POSITIVE},
+            {"id": "pos-2", "diagnosis_result": COVID_RESPONSE_POSITIVE},
+            {"id": "neg-1", "diagnosis_result": COVID_RESPONSE_NEGATIVE},
+            {"id": "neg-2", "diagnosis_result": COVID_RESPONSE_NEGATIVE},
+            {"id": "cov-1",
+                "diagnosis_result": COVID_RESPONSE_POSITIVE},
+            {"id": "cov-2",
+                "diagnosis_result": COVID_RESPONSE_POSITIVE},
+            {"id": "cov-3", "diagnosis_result": FAILED_BY_TOO_HIGH_COVID_VALUE}]
+
+        assert expected == list(result)
+
+    def test_can_analyze_samples_scenario8(self):
```

Comment:
Perhaps there should be a test with positive control have fam = 0 and hex =0, and all samples are failed?

Reply:
Will add one.